python/lsst/afw/geom/testUtils.py

```diff
@@ -91,3 +100,759 @@ def __iter__(self):
+                        transform2 = transform1Class(polyMap)
+                        with self.assertRaises(InvalidParameterError):
+                            transform = transform2.of(transform1)
+
+    def checkOfChaining(self):
+        """Test that both conventions for chaining Transform*To*.of give
+        the same result
+        """
+        transform1 = afwGeom.TransformGenericToGeneric(
+            makeForwardPolyMap(2, 3))
+        transform2 = afwGeom.TransformGenericToGeneric(
+            makeForwardPolyMap(3, 4))
+        transform3 = afwGeom.TransformGenericToGeneric(
+            makeForwardPolyMap(4, 1))
+
+        merged1 = transform3.of(transform2).of(transform1)
+        merged2 = transform3.of(transform2.of(transform1))
+
+        fromEndpoint = transform1.getFromEndpoint()
+        toEndpoint = transform3.getToEndpoint()
+
+        inPoint = fromEndpoint.pointFromData(self.makeRawPointData(2))
+        assert_allclose(toEndpoint.dataFromPoint(merged1.tranForward(inPoint)),
+                        toEndpoint.dataFromPoint(merged2.tranForward(inPoint)))
+
+    def checkPersistence(self, transform):
+        """Check persistence of a transform
+        """
+        className = type(transform).__name__
+        fileName = "persisted_{}.dat".format(className)
+        filePath = os.path.join(self.getTestDir(), fileName)
+        transform.saveToFile(filePath)
+        transformRoundTrip = afwGeom.readTransform(filePath)
+        self.assertEqual(type(transform), type(transformRoundTrip))
+        self.assertEqual(transform.getFrameSet().show(),
+                         transformRoundTrip.getFrameSet().show())
+
+        fromEndpoint = transform.getFromEndpoint()
+        toEndpoint = transform.getToEndpoint()
+        frameSet = transform.getFrameSet()
+        nIn = frameSet.getNin()
+        nOut = frameSet.getNout()
+
+        if frameSet.hasForward():
+            nPoints = 7  # arbitrary
+            rawInArray = self.makeRawArrayData(nPoints, nIn)
+            inArray = fromEndpoint.arrayFromData(rawInArray)
+            outArray = transform.tranForward(inArray)
+            outData = toEndpoint.dataFromArray(outArray)
+            outArrayRoundTrip = transformRoundTrip.tranForward(inArray)
+            outDataRoundTrip = toEndpoint.dataFromArray(outArrayRoundTrip)
+            assert_allclose(outData, outDataRoundTrip)
+
+        if frameSet.hasInverse():
+            nPoints = 7  # arbitrary
+            rawOutArray = self.makeRawArrayData(nPoints, nOut)
+            outArray = toEndpoint.arrayFromData(rawOutArray)
+            inArray = transform.tranInverse(outArray)
+            inData = fromEndpoint.dataFromArray(inArray)
+            inArrayRoundTrip = transformRoundTrip.tranInverse(outArray)
+            inDataRoundTrip = fromEndpoint.dataFromArray(inArrayRoundTrip)
+            assert_allclose(inData, inDataRoundTrip)
+        os.remove(filePath)
```

Comment:
I intentionally made it not `finally` so the file would be left behind on test failure. I'll add a code comment.